pkg/session.go

```diff
@@ -0,0 +1,80 @@
+package cluster
+
+import (
+	"sync"
+
+	sfu "github.com/pion/ion-sfu/pkg/sfu"
+)
+
+type Broadcast struct {
+	method string
+	params interface{}
+}
+
+type Session struct {
+	mu               sync.Mutex
+	presence         map[string]interface{}
+	presenceRevision uint64
+
+	broadcastListeners map[string]chan<- Broadcast
+
+	sfu.SessionLocal
+}
+
+func NewSession(id string, dcs []*sfu.Datachannel, cfg sfu.WebRTCTransportConfig) Session {
+	return Session{
+		sync.Mutex{},
+		make(map[string]interface{}),
+		0,
+		make(map[string]chan<- Broadcast),
+		*sfu.NewSession(id, dcs, cfg).(*sfu.SessionLocal),
+	}
+}
+
+func (s *Session) UpdatePresenceMetaForPeer(peerID string, meta interface{}) {
+	s.mu.Lock()
+	s.mu.Unlock()
+
+	if meta != nil {
+		s.presence[peerID] = meta
+		s.presenceRevision += 1
```

Comment:
nit: since revision is being bumped anyway, can be out of if

Reply:
👍